CODE/Tests/Fake/FakeAWSHTTPClient.swift

```diff
@@ -0,0 +1,36 @@
+import SotoCore
+import NIOCore
+import NIOHTTP1
+import Logging
+import Foundation
+
+public class FakeAWSHTTPClient: AWSHTTPClient {
```

Comment:
Isn't this protocol disappearing?

Reply:
o hmm, I don't follow AWS SDKs closely, but I can see that they're removing it: https://github.com/soto-project/soto-core/commit/addcfe6fd27b05e5fedb83ebbd29d90223100432

I think we might need to make a "AWSClientProtocol" of ourselves now?

Reply:
oo the thing is we're still using soto 5 instead of the newest soto 6 ... that's why they are removing the protocol (since it no longer has any use in soto 6)

Reply:
We can update Soto in a separate PR. Where are you abstracting the AWSClient? (I'm on mobile). We probably want to move the abstraction a layer up rather to eg a EC2Interface rather than the client directly. That will make it easier to migrate Soto versions too

Reply:
So that'll be a "AWSClientProtocol" of ourselves again? This way we can hide the AWS interface under a protocol of ourselves and only add functions we need (which makes updating easier too)

Reply:
Yes though I think mocking at the client level is too low and it should be abstracted higher. (Think the UserRepository level instead of mocking DynamoDB directly)

Reply:
about "where?" its in the "add coin lambda handler" (PennyAPI/AddCoin)

Reply:
The AWSClient is also used in `PennySHARED/Services` (btw we should restructure the folders and targets too 😅 in another pr)

So should i create a new target "AWSClientWrapper"?

Reply:
Sorry it appears Github doesn't update the conversation unless you refresh the page, so i missed your last comment.

Reply:
Replying to your last comment:
Yeah i was thinking of that too a while ago (although i forgot about it later)...  i'll see what i can do

Reply:
I didn't find a better solution than the `underlyingHandler` solution.
We can simplify these _after_ we update our soto stuff, including the runtime.

Reply:
Ok i just noticed something ... let me check ...

Reply:
Pushed some more changes to simplify those stuff